src/Microsoft.PowerShell.Commands.Utility/commands/utility/ConvertTo-Html.cs

```diff
@@ -647,6 +669,40 @@ private void WritePropertyValue(StringBuilder Listtag, MshParameter p)
             }
         }
 
+        /// <summary>
+        /// Check if a string or substrings in a string is uri, if true, then change to hyperlink format.
+        /// </summary>
+        /// <param name="s"> the string that needs to be altered </param>
+        /// <param name="hyper"> if true, then change to hyperlink format </param>
+        /// <returns>the midified string/unmodified string.</returns>
+        private static string CheckUri(string s, bool hyper)
+        {
+            if (hyper)
+            {
+                Regex regHttp = new Regex(@"(?<=\()\b(https?://|www\.)[-A-Za-z0-9+&@#/%?=~_()|!:,.;]*[-A-Za-z0-9+&@#/%=~_()|](?=\))|(?<=(?<wrap>[=~|_#]))\b(https?://|www\.)&[-A-Za-z0-9+&@#/%?=~_()|!:,.;]*[-A-Za-z0-9+&@#/%=~_()|](?=\k<wrap>)|\b(https?://|www\.)[-A-Za-z0-9+&@#/%?=~_()|!:,.;]*[-A-Za-z0-9+&@#/%=~_()|]", RegexOptions.Compiled | RegexOptions.IgnoreCase);
```

Comment:
There may be an issue with this regular expression. I'm not sure it will capture something like 
`https://www.t1.bar http://www.t2.bar` correctly. Should it capture 2 uris?

Also, I have a feeling the initialization of the regex should happen only once (so it should probably be a class member and static. The way it is now the regex will be created a bunch of times. (and since it's compiled it will create a new assembly each time).